torch/csrc/jit/passes/onnx/scalar_type_analysis.cpp

```diff
@@ -175,10 +175,10 @@ static void UpdateScalarTypeForInputs(
 
   for (auto input : n->inputs()) {
     auto input_tensor_type = input->type()->cast<TensorType>();
-    auto input_scalar_type = input_tensor_type->scalarType();
```

Comment:
consider putting the below check on `input_tensor_type` here. And handle what to do when `input_tensor_type` is not available.

On the other hand, is it `ListType` input in this case? do we need to take into consideration its scalar type as well?